src/HTML5/Parser/Scanner.php

```diff
@@ -61,6 +61,30 @@ public function __construct($data, $encoding = 'UTF-8')
         $this->EOF = strlen($data);
     }
 
+    /**
+     * Check if upcomming chars match the given sequence.
+     *
+     * This will read the stream for the $sequence. If it's
+     * found, this will return true. If not, return false.
+     * Since this unconsumes any chars it reads, the caller
+     * will still need to read the next sequence, even if
+     * this returns true.
+     *
+     * Example: $this->scanner->sequenceMatches('</script>') will
+     * see if the input stream is at the start of a
+     * '</script>' string.
+     *
+     * @param string $sequence
+     * @param bool $caseSensitive
+     *
+     * @return bool
+     */
+    public function sequenceMatches($sequence, $caseSensitive = true)
+    {
+        $portion = substr($this->data, $this->char, strlen($sequence));
```

Comment:
Wouldn't using `mb_*` functions be safer for UTF-8/16 strings?

Reply:
thats true!

Reply:
Not sure about it, as this is used to lookup for html tags, that are always ascii (and mb_*) functions are slower.

If that was the case, most of the functions in the scanner and tokenizer will be broken

Reply:
As long as you realize you are working on bytes and not characters, using the plain `str*` functions is fine with UTF-8 encoded strings (and much faster).

In `php-typography`, I do [a lookup](https://github.com/mundschenk-at/php-typography/blob/master/src/class-strings.php) whether a given string (`DOMText` content) contains UTF-8 characters and choose the appropriate function that way. However, that is mainly necessary for determining whether the `u` flag for regular expressions needs to be used.

Reply:
(Lookup and replacement of ASCII sequences should be "UTF-8 safe" as no multibyte sequence uses ASCII characters. Be careful with `preg_*`, though, as I've had PCRE generate invalid sequences when operating on certain multibyte characters and not using the `u` modifier.)

Reply:
In this specific case, it does not matter as we are looking for a specific string, so `substr` + `strlen` and `===` comparison will work. The issue might occur in case of case-insensitve comparisons.

Reply:
As long as you are looking for an ASCII sequence, that should still be fine.